whatis277/harvest
# trellis: Add new router

Adds new custom router built from nginx

Trellis helps build a custom image, which we'll use for production later

Moves services outside the app context and into trellis
Applications shouldn't be aware of the reverse proxy

Improves the documentation on the nginx config as well

Also adds a resolver which ignores the config until upstream is available

Finally, replaces `latest` with versions in compose
This will be helpful for production later

Testing instructions:
1. `dc up --build trellis`

1. Wait a couple of seconds until trellis is up and running 

1. Ensure there's no "upstream not found" errors in the logs

1. `dc up bean`

1. Wait a couple of seconds until bean is up and running

1. Ensure http://whatisbean.local/ loads just fine 

1. `dc up pear`

1. Wait a couple of seconds until pear is up and running

1. Ensure http://whatispear.local/ loads just fine
